src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -46,121 +46,146 @@
-        {
-            EmberStatus status;
-            binding.type    = EMBER_MULTICAST_BINDING;
-            binding.groupId = groupId;
-            binding.local   = endpoint;
+    GroupDataProvider * provider = GetGroupDataProvider();
+    VerifyOrReturnError(nullptr != provider, EMBER_ZCL_STATUS_NOT_FOUND);
+    GroupDataProvider::GroupMapping mapping(endpointId, groupId, groupName);
 
-            status = emberSetBinding(i, &binding);
-            if (status == EMBER_SUCCESS)
-            {
-                // Set the group name, if supported
-                emberAfPluginGroupsServerSetGroupNameCallback(endpoint, groupId, groupName);
-                return EMBER_ZCL_STATUS_SUCCESS;
-            }
-            else
-            {
-                emberAfGroupsClusterPrintln("ERR: Failed to create binding (0x%x)", status);
-            }
-        }
+    CHIP_ERROR err = provider->AddGroupMapping(fabricIndex, mapping);
+    if (CHIP_NO_ERROR == err)
+    {
+        return EMBER_ZCL_STATUS_SUCCESS;
+    }
+    else
+    {
+        ChipLogDetail(Zcl, "ERR: Failed to add mapping (end:%d, group:0x%x), err:%" CHIP_ERROR_FORMAT, endpointId, groupId,
+                      err.Format());
+        return EMBER_ZCL_STATUS_INSUFFICIENT_SPACE;
     }
-    emberAfGroupsClusterPrintln("ERR: Binding table is full");
-    return EMBER_ZCL_STATUS_INSUFFICIENT_SPACE;
 }
 
-static EmberAfStatus removeEntryFromGroupTable(EndpointId endpoint, GroupId groupId)
+static EmberAfStatus GroupRemove(FabricIndex fabricIndex, EndpointId endpointId, GroupId groupId)
 {
-    if (isGroupPresent(endpoint, groupId))
+    VerifyOrReturnError(IsFabricGroupId(groupId), EMBER_ZCL_STATUS_INVALID_VALUE);
+    VerifyOrReturnError(GroupExists(fabricIndex, endpointId, groupId), EMBER_ZCL_STATUS_NOT_FOUND);
+
+    GroupDataProvider * provider = GetGroupDataProvider();
+    VerifyOrReturnError(nullptr != provider, EMBER_ZCL_STATUS_NOT_FOUND);
+    GroupDataProvider::GroupMapping mapping(endpointId, groupId);
+
+    CHIP_ERROR err = provider->RemoveGroupMapping(fabricIndex, mapping);
+    if (CHIP_NO_ERROR == err)
     {
-        uint8_t bindingIndex = findGroupIndex(endpoint, groupId);
-        EmberStatus status   = emberDeleteBinding(bindingIndex);
-        if (status == EMBER_SUCCESS)
-        {
-            emberAfPluginGroupsServerSetGroupNameCallback(endpoint, groupId, CharSpan());
-            return EMBER_ZCL_STATUS_SUCCESS;
-        }
-        else
-        {
-            emberAfGroupsClusterPrintln("ERR: Failed to delete binding (0x%x)", status);
-            return EMBER_ZCL_STATUS_FAILURE;
-        }
+        return EMBER_ZCL_STATUS_SUCCESS;
+    }
+    else
+    {
+        ChipLogDetail(Zcl, "ERR: Failed to remove mapping (end:%d, group:0x%x), err:%" CHIP_ERROR_FORMAT, endpointId, groupId,
+                      err.Format());
+        return EMBER_ZCL_STATUS_FAILURE;
     }
+
     return EMBER_ZCL_STATUS_NOT_FOUND;
```

Comment:
I don't understand this part: that return statement is never reached.  How are we implementing the spec requirement around returning a NOT_FOUND error if the group we are removing is not there to be removed?